cdk/src/components/privacy-notification-function/privacy-notification-function.ts

```diff
@@ -0,0 +1,70 @@
+import * as cdk from 'aws-cdk-lib'
+import { Construct } from 'constructs'
+import * as lambda from 'aws-cdk-lib/aws-lambda'
+import * as nodejsLambda from 'aws-cdk-lib/aws-lambda-nodejs'
+import * as path from 'node:path'
+import * as events from 'aws-cdk-lib/aws-events'
+import * as targets from 'aws-cdk-lib/aws-events-targets'
+import * as ec2 from 'aws-cdk-lib/aws-ec2'
+import * as ssm from 'aws-cdk-lib/aws-ssm'
+import * as iam from 'aws-cdk-lib/aws-iam'
+
+// Define the properties for the PrivacyNotificationFunction construct
+export interface PrivacyNotificationFunctionProps {
+  vpc: ec2.Vpc
+}
+
+export class PrivacyNotificationFunction extends Construct {
+  constructor(
+    scope: Construct,
+    id: string,
+    props: PrivacyNotificationFunctionProps
+  ) {
+    super(scope, id)
+
+    const { vpc } = props
+
+    // Create a secret in SSM Parameter Store using the env variable
+    const ssmSecret = new ssm.StringParameter(this, 'WebhookUrl', {
+      parameterName: '/privacy-notification-function/webhook-url',
```

Comment:
Can we lift this to a prop? That way we can resolve it from the stack and pass in the SSM parameter name. We can use [`getSecret`](https://github.com/aws-amplify/discord-bot/blob/main/cdk/src/stack.ts#LL177C11-L177C20) to retrieve the SSM Parameter construct by it's name (e.g. `WEBHOOK_URL`) and pass in